src/io-methods/default.js

```diff
@@ -1,30 +1,26 @@
 "use strict";
 const _ = require("lodash");
-const fs = require("./filesystem");
+const filesystem = require("./filesystem");
 const s3 = require("./s3");
 const azure = require("./azure");
 const google = require("./google-cloud-storage");
-const config = require("../config").config;
+const { getConfig } = require("../config");
 const memory = require("./memory");
 
-const defaultFilePath =
-  config && config.manifestFormat === "importmap"
-    ? "import-map.json"
-    : "sofe-manifest.json";
-
 function getFilePath(env) {
+  const config = getConfig();
   if (_.has(config, ["locations", env])) {
     return config.locations[env];
   } else if (_.has(config, ["locations", "default"])) {
     return config.locations.default;
-  } else if (!_.isEmpty(config.locations)) {
+  } else if (!env && !_.isEmpty(config.locations)) {
     return config.locations[Object.keys(config.locations)[0]];
   } else {
-    return defaultFilePath;
+    throw Error(`No such environment '${env}'. Check your configuration file.`);
```

Comment:
This is the breaking change - I've always found it confusing that instead of sending 404 it would send you an empty import map from a default file. This is the breaking change part, but i think will be a better DX.